lib/cocoapods-binary-cache/pod-rome/xcodebuild_command.rb

```diff
@@ -81,16 +93,56 @@ def build_for_sdk(sdk)
       )
     end
 
+    def archive_for_sdk(sdk, target)
+      PodPrebuild::XcodebuildCommand.xcodebuild_archive(
+        sandbox: sandbox,
+        target: target,
+        configuration: configuration,
+        sdk: sdk,
+        build_dir: build_dir,
+        args: sdk == simulator ? @build_args[:simulator] : @build_args[:device]
+      )
+    end
+
     def create_xcframework(target)
-      non_framework_paths = Dir[target_products_dir_of(target, sdks[0]) + "/*"] - [framework_path_of(target, sdks[0])]
+      non_framework_paths = Dir[target_products_dir_of(target, sdks[0]) + "/*"] \
+        - [ framework_path_of(target, sdks[0]),         # prevent collect .framework
+            framework_product_path_of(target, sdks[0]),    # prevent collect .framework
+            dsym_path_of(target, sdks[0]),              # prevent collect dSYMs
+            bcsymbolmap_path_of(target, sdks[0]) ]      # prevent collect BCSymbolMaps
       collect_output(target, non_framework_paths)
 
       output = "#{output_path(target)}/#{target.product_module_name}.xcframework"
       FileUtils.rm_rf(output)
 
-      cmd = ["xcodebuild", " -create-xcframework", "-allow-internal-distribution"]
-      cmd += sdks.map { |sdk| "-framework #{framework_path_of(target, sdk)}" }
+      cmd = ["xcodebuild", "-create-xcframework", "-allow-internal-distribution"]
+
+      # for each sdk, the order of params must be -framwork then -debug-symbols
+      # to prevent duplicated file error when copying dSYMs
+      sdks.each do |sdk|
+        cmd << "-framework" << framework_path_of(target, sdk)
+
+        dsym_path = dsym_path_of(target, sdk)
+        if Dir.exists?(dsym_path)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/grab/cocoapods-binary-cache/blob/b144f30fb41bd34e49fceb072710e5f2c3a3e710/lib/cocoapods-binary-cache/pod-rome/xcodebuild_command.rb#L126"></span>Lint/DeprecatedClassMethods: <code>Dir.exists?</code> is deprecated in favor of <code>Dir.exist?</code>.</td>
    </tr>
  </tbody>
</table>